dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +53,176 @@ public Pattern getParameterPattern() {
         return paramRegex;
     }
 
-    public Map<String, DbParameterAccessor> getAllColumns(
-            final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
-                + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
-                + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+    public Map<String, DbParameterAccessor> getAllColumns(String tableName)
+                                                        throws SQLException {
+        DatabaseTable t = new DatabaseTable(getConnection(), getConnection().getSchema(), tableName.toUpperCase());
+        return t.getParams();
     }
 
-    private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
-        try (PreparedStatement dc = getConnection().prepareStatement(query)) {
-            dc.setString(1, tableOrViewName);
+    private abstract class DatabaseObject {
+        protected Connection con;
```

Comment:
Hmm, we have many names and abbreviations for the connection (con, conn, ...).

Technically here we have a direct access here to the parent `getConnection()` of DerbyEnvironment (since it's a non-static inner class of it).